@here/harp-mapview/test/PlacementTest.ts

```diff
@@ -744,10 +753,316 @@ describe("Placement", function() {
                     true
                 );
                 // Label placed and layout changed diagonally.
-                layout = textElement.layoutStyle!;
+                layout = state.textLayoutState!;
+                expect(result).to.equal(PlacementResult.Ok);
+                expect(layout.horizontalAlignment).to.be.equal(HorizontalAlignment.Left);
+                expect(layout.verticalAlignment).to.be.equal(VerticalAlignment.Below);
+            });
+
+            it("places text moving to the center-left screen side", async function() {
+                const textElement = await createTextElement(
+                    textCanvas,
+                    "Test Horizontal Alternative",
+                    new THREE.Vector3(),
+                    {},
+                    {
+                        horizontalAlignment: HorizontalAlignment.Right,
+                        verticalAlignment: VerticalAlignment.Center,
+                        wrappingMode: WrappingMode.None
+                    }
+                );
+                const state = createTextElementState(textElement);
+
+                const outPosition = new THREE.Vector3();
+                const inPosition = new THREE.Vector2(-10, 0).add(screenCenterLeft);
+
+                // Set the current style for the canvas.
+                textCanvas.textRenderStyle = textElement.renderStyle!;
+                textCanvas.textLayoutStyle = textElement.layoutStyle!;
+                // Run placement without multi-anchor support.
+                let result = placePointLabel(
+                    state,
+                    inPosition,
+                    1.0,
+                    textCanvas,
+                    screenCollisions,
+                    false,
+                    outPosition,
+                    false
+                );
+                // Label out of screen and layout unchanged.
+                let layout = state.textLayoutState!;
+                expect(result).to.equal(PlacementResult.Invisible);
+                expect(layout.horizontalAlignment).to.be.equal(HorizontalAlignment.Right);
+                expect(layout.verticalAlignment).to.be.equal(VerticalAlignment.Center);
```

Comment:
This code is repeated many many times along this file (the file is already almost 1500 lines!), making it hard to maintain. Any change in createTextElement, createTextElementState or placePointLabel parameters will have to be applied to all those places.

For these test cases where you use the same setup/check code over and over again, it's very useful to use a parameterized test pattern. You have a test that takes as input some configuration parameters (in this case the alignment alternatives and the labels anchor positions) and the expected outcome (in this case the placement result), and executes the same code.

See for example intersectsRay tests OrientedBox3Test.

Reply:
Will do in the next update.

Reply:
Done.